src/ai/backend/manager/container_registry/base.py

```diff
@@ -128,13 +129,15 @@ async def rescan_single_registry(
                 self.credentials["password"] = password
             async with self.prepare_client_session() as (url, client_session):
                 self.registry_url = url
-                async with aiotools.TaskGroup() as tg:
-                    try:
-                        async for image in self.fetch_repositories(client_session):
-                            tg.create_task(self._scan_image(client_session, image))
-                    except Exception as e:
-                        error_msg = f"Failed to fetch repositories! (registry: {self.registry_name}, project: {self.registry_info.project}). Detail: {str(e)}"
-                        log.error(error_msg)
+
+                tasks = []
+                async for image in self.fetch_repositories(client_session):
+                    task = asyncio.create_task(self._scan_image(client_session, image))
+                    tasks.append(task)
+
+                for res in await asyncio.gather(*tasks, return_exceptions=True):
```

Comment:
In this case, asyncio.as_completed seems more appropriate.

Reply:
We decided to use `asyncio.gather()` to prevent exception handling from becoming too complex.